### PR TITLE
allows med borgs to pet nymphs and mice

### LINKED ISSUE #18895: Medical Cyborgs cannot drop simplemobs they pick up in their help intent module
**Issue Description**:
Medical cyborgs can both pick up simplemobs with their help intent module and are unable to drop them

**What did you expect to happen**: 
To either not have this happen in the first place, or have the ability to drop the simplemob

**What happened instead**:
You can pick up the rat but not get rid of it

**Why is this bad/What are the consequences**:
rat STUCK. Sentient simplemobs can get out by resisting, but not the stupid ones

**Steps to reproduce the problem**:
Use your hugging module on a small simplemob
yoink

**When did the problem start happening**:
Likely since this module got added

**Extra information**:
oh shit a rat

### PR DESCRIPTION
**Pull request by tsp01**
## What Does This PR Do
Technically fixes #18895
This PR prevents medical borgos from picking up mice or diona nymphs. Now they pet them instead. This used to put them into the borg and they could not be dropped.

## Why It's Good For The Game
No more undroppable mice

## Testing
1. Spawn as med borg
2. pet mouse
3. pet nymph
4. profit!

## Changelog
:cl:
fix: Fixed being able to pick up mice as medical borg
/:cl:

### CODE REVIEW

**Comment by lewcc** [comment]
I think you might be able to check `holder_type` on the mobs themselves, which should keep this same behavior for anything else small that can be picked up

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -72,7 +72,7 @@
 	switch(mode)
 		if(CYBORG_HUGS)
 			if(M.health >= 0)
-				if(isanimal(M))
+				if(isanimal(M) && !M.holder_type)
```
Comment (Sirryan2002) [inline code/game/objects/items/robot/robot_items.dm:75]: can you add a comment here explaining why `M.holder_type` is used here so people don't have to go code diving to infer purpose?

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — CHANGES_REQUESTED** [review]
Please do as sirryan requested before merge, good documentation helps poor sods in the future that have to touch this code

File: code/game/objects/items/robot/robot_items.dm (hunk 1 above)
Comment (tsp01) [reply code/game/objects/items/robot/robot_items.dm]: Done!

**Review by S34NW — APPROVED** [review]

**Pull request merged**